broker/src/main/java/io/camunda/zeebe/broker/clustering/topology/ClusterTopologyManager.java

```diff
@@ -0,0 +1,103 @@
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Responsible for initializing ClusterTopology and managing ClusterTopology changes. */
+final class ClusterTopologyManager {
+  private static final Logger LOG = LoggerFactory.getLogger(ClusterTopologyManager.class);
+
+  private final ConcurrencyControl executor;
+  private final PersistedClusterTopology persistedClusterTopology;
+
+  ClusterTopologyManager(
+      final ConcurrencyControl executor, final PersistedClusterTopology persistedClusterTopology) {
+    this.executor = executor;
+    this.persistedClusterTopology = persistedClusterTopology;
+  }
+
+  ActorFuture<ClusterTopology> getClusterTopology() {
+    return executor.call(persistedClusterTopology::getTopology);
+  }
+
+  ActorFuture<Void> start(final BrokerCfg brokerCfg) {
+    final ActorFuture<Void> startFuture = executor.createFuture();
+
+    executor.run(
+        () -> {
+          try {
+            initialize(brokerCfg);
+            startFuture.complete(null);
+          } catch (final Exception e) {
+            LOG.error("Failed to initialize ClusterTopology", e);
+            startFuture.completeExceptionally(e);
+          }
+        });
+
+    return startFuture;
+  }
+
+  private void initialize(final BrokerCfg brokerCfg) throws IOException {
+    persistedClusterTopology.initialize();
+    if (persistedClusterTopology.getTopology() == null) {
+      final var topology = initializeFromConfig(brokerCfg);
+      persistedClusterTopology.update(topology);
+      LOG.trace(
+          "Initialized ClusterTopology from BrokerCfg {}", persistedClusterTopology.getTopology());
+    }
+  }
+
+  private ClusterTopology initializeFromConfig(final BrokerCfg brokerCfg) throws IOException {
+    final var partitionDistribution =
+        new StaticPartitionDistributionResolver()
+            .resolveTopology(brokerCfg.getExperimental().getPartitioning(), brokerCfg.getCluster());
+
+    final var partitionsOwnedByMembers =
+        partitionDistribution.partitions().stream()
+            .flatMap(
+                p ->
+                    p.members().stream()
+                        .map(m -> Map.entry(m, Map.entry(p.id().id(), p.getPriority(m)))))
+            .collect(
+                Collectors.groupingBy(
+                    Entry::getKey,
+                    Collectors.toMap(
+                        e -> e.getValue().getKey(),
+                        e -> PartitionState.active(e.getValue().getValue()))));
+
+    final var memberStates =
+        partitionsOwnedByMembers.entrySet().stream()
+            .map(
+                entry -> {
+                  final MemberId memberId = entry.getKey();
+                  final Map<Integer, PartitionState> partitionInfo = entry.getValue();
+                  return Map.entry(memberId, MemberState.initializeAsActive(partitionInfo));
+                })
+            .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
```

Comment:
🙃️ `.collect(Collectors.toMap(Entry::getKey, (e) -> MemberState.initializeAsActive(e.getValue()))` would be enough, no need for the intermediate `map`